From 2f25ce253ef4dec015ed06c54253df8b361b834b Mon Sep 17 00:00:00 2001
From: Roman Beranek <roman.beranek@prusa3d.com>
Date: Wed, 18 Aug 2021 14:35:53 +0200
Subject: [PATCH] autopilot-rs offline

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index f5bee86..aac01b8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,7 +12,7 @@ url = "^2.2"
 tokio = { version = "0.2", features = ["full"] }
 handlebars = "^3.5"
 websocket = "^0.26"
-autopilot = { git = "https://github.com/H-M-H/autopilot-rs.git", rev = "63eed09c715bfb665bb23172a3930a528e11691c" }
+autopilot = "0.4.0"
 serde = { version = "^1.0", features = ["derive"] }
 serde_json = "^1.0"
 tracing = "^0.1"
-- 
2.32.0

