diff --git a/xs/CMakeLists.txt b/xs/CMakeLists.txt
index 9e23112..97bccf8 100644
--- a/xs/CMakeLists.txt
+++ b/xs/CMakeLists.txt
@@ -531,20 +531,20 @@ endif ()
 include_directories(${GLEW_INCLUDE_DIRS})
 target_link_libraries(XS ${GLEW_LIBRARIES})
 # Install the XS.pm and XS.{so,dll,bundle} into the local-lib directory.
-set(PERL_LOCAL_LIB_DIR "${PROJECT_SOURCE_DIR}/local-lib/lib/perl5/${PerlEmbed_ARCHNAME}")
+set(PERL_LIB_DIR "${PROJECT_SOURCE_DIR}/lib/perl5/${PerlEmbed_ARCHNAME}")
 add_custom_command(
     TARGET XS
     POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E make_directory "${PERL_LOCAL_LIB_DIR}/auto/Slic3r/XS/"
-        COMMAND ${CMAKE_COMMAND} -E copy "$<TARGET_FILE:XS>" "${PERL_LOCAL_LIB_DIR}/auto/Slic3r/XS/"
-        COMMAND ${CMAKE_COMMAND} -E make_directory "${PERL_LOCAL_LIB_DIR}/Slic3r/"
-        COMMAND ${CMAKE_COMMAND} -E copy "${PROJECT_SOURCE_DIR}/xs/lib/Slic3r/XS.pm" "${PERL_LOCAL_LIB_DIR}/Slic3r/"
+        COMMAND ${CMAKE_COMMAND} -E make_directory "${PERL_LIB_DIR}/auto/Slic3r/XS/"
+        COMMAND ${CMAKE_COMMAND} -E copy "$<TARGET_FILE:XS>" "${PERL_LIB_DIR}/auto/Slic3r/XS/"
+        COMMAND ${CMAKE_COMMAND} -E make_directory "${PERL_LIB_DIR}/Slic3r/"
+        COMMAND ${CMAKE_COMMAND} -E copy "${PROJECT_SOURCE_DIR}/xs/lib/Slic3r/XS.pm" "${PERL_LIB_DIR}/Slic3r/"
 )
 if(APPLE)
     add_custom_command(
         TARGET XS
         POST_BUILD
-            COMMAND ${CMAKE_COMMAND} -E rename "${PERL_LOCAL_LIB_DIR}/auto/Slic3r/XS/XS" "${PERL_LOCAL_LIB_DIR}/auto/Slic3r/XS/XS.bundle"
+            COMMAND ${CMAKE_COMMAND} -E rename "${PERL_LIB_DIR}/auto/Slic3r/XS/XS" "${PERL_LIB_DIR}/auto/Slic3r/XS/XS.bundle"
     )
 endif()
 
