diff -Nur a/perf-5.11.ebuild b/perf-5.11.ebuild
--- a/perf-5.11.ebuild	2021-02-15 16:09:07.000000000 +0100
+++ b/perf-5.11.ebuild	2021-01-02 00:09:13.694217864 +0100
@@ -50,8 +50,8 @@
 RDEPEND="audit? ( sys-process/audit )
 	crypt? ( dev-libs/openssl:0= )
 	clang? (
-		<sys-devel/clang-10:*
-		<sys-devel/llvm-10:*
+		<sys-devel/clang-12:*
+		<sys-devel/llvm-12:*
 	)
 	demangle? ( sys-libs/binutils-libs:= )
 	gtk? ( x11-libs/gtk+:2 )
@@ -86,7 +86,7 @@
 }
 
 pkg_setup() {
-	use clang && LLVM_MAX_SLOT=9 llvm_pkg_setup
+	use clang && LLVM_MAX_SLOT=11 llvm_pkg_setup
 }
 
 src_unpack() {
@@ -110,6 +110,10 @@
 		eshopts_pop
 	fi
 
+	pushd "${S_K}" >/dev/null || die
+	eapply "${FILESDIR}/${P}-fix-clang-feature-test.patch"
+	popd || die
+
 	local a
 	for a in ${A}; do
 		[[ ${a} == ${LINUX_SOURCES} ]] && continue
