diff --git a/setup.py b/setup.py
index aa1bb78..5d255b1 100644
--- a/setup.py
+++ b/setup.py
@@ -140,7 +140,7 @@ setup(
     package_dir={'pyqtgraph.examples': 'examples'},  ## install examples along with the rest of the source
     package_data={'pyqtgraph.examples': ['optics/*.gz', 'relativity/presets/*.cfg']},
     install_requires = [
-        'numpy>=1.8.0',
+        'numpy (>=1.8.0)',
         ],
     **setupOpts
 )
