From ff77dd8a1ffd2d027c63386ca5622fbc66d426f9 Mon Sep 17 00:00:00 2001
From: Root <root@gentoo.local>
Date: Thu, 13 Aug 2020 08:00:51 +0200
Subject: [PATCH] fix clang feature test

---
 tools/build/feature/Makefile | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/tools/build/feature/Makefile b/tools/build/feature/Makefile
index b1f0321..fc81789 100644
--- a/tools/build/feature/Makefile
+++ b/tools/build/feature/Makefile
@@ -282,7 +282,7 @@ $(OUTPUT)test-sdt.bin:
 	$(BUILD)
 
 $(OUTPUT)test-cxx.bin:
-	$(BUILDXX) -std=gnu++11
+	$(BUILDXX) -std=gnu++14
 
 $(OUTPUT)test-gettid.bin:
 	$(BUILD)
@@ -307,12 +307,11 @@ $(OUTPUT)test-llvm-version.bin:
 		> $(@:.bin=.make.output) 2>&1
 
 $(OUTPUT)test-clang.bin:
-	$(BUILDXX) -std=gnu++11 				\
+	$(BUILDXX) -std=gnu++14 				\
 		-I$(shell $(LLVM_CONFIG) --includedir) 		\
 		-L$(shell $(LLVM_CONFIG) --libdir)		\
-		-Wl,--start-group -lclangBasic -lclangDriver	\
-		  -lclangFrontend -lclangEdit -lclangLex	\
-		  -lclangAST -Wl,--end-group 			\
+		-Wl,--start-group -lclang -lclang-cpp		\
+		-Wl,--end-group 				\
 		$(shell $(LLVM_CONFIG) --libs Core option)	\
 		$(shell $(LLVM_CONFIG) --system-libs)		\
 		> $(@:.bin=.make.output) 2>&1
-- 
2.28.0

