diff --git a/configure.ac b/configure.ac
index 025bda7..4ec08c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -305,9 +305,9 @@ if test "${enable_ncurses}" != "no"; then
   AC_CHECK_HEADERS(ncursesw/ncurses.h ncurses/ncurses.h ncurses.h curses.h,
    [ncurses="no"
     AC_CHECK_LIB(ncursesw, initscr,
-     [ncurses="-lncursesw"],
+     [ncurses="-lncursesw -ltinfow"],
      [AC_CHECK_LIB(ncurses, initscr,
-       [ncurses="-lncurses"])])
+       [ncurses="-lncurses -ltinfo"])])
     if test "${ncurses}" = "no"; then
       continue
     fi
