diff --git a/src/TerminalDisplay.cpp b/src/TerminalDisplay.cpp
index 960cb9d2..2132dde4 100644
--- a/src/TerminalDisplay.cpp
+++ b/src/TerminalDisplay.cpp
@@ -2016,8 +2016,8 @@ void TerminalDisplay::mousePressEvent(QMouseEvent* ev)
             // No reason to ever start a drag event
             _dragInfo.state = diNone;
 
-            _preserveLineBreaks = !((ev->modifiers() & Qt::ControlModifier) && !(ev->modifiers() & Qt::AltModifier));
-            _columnSelectionMode = (ev->modifiers() & Qt::AltModifier) && (ev->modifiers() & Qt::ControlModifier);
+            _preserveLineBreaks = true;
+            _columnSelectionMode = (ev->modifiers() & Qt::ControlModifier);
 
             if (_mouseMarks || (ev->modifiers() == Qt::ShiftModifier)) {
                 // Only extend selection for programs not interested in mouse
